conformance/lang/expressions/unary-expression/unary-complement-expression.balt

```diff
@@ -0,0 +1,483 @@
+
+public function main() {
+    int:Unsigned8 unsignedInt = 86;
+    int res = ~unsignedInt;
+    io:println(res); // @output -87
+
+    io:println(~int:UNSIGNED8_MAX_VALUE);  // @output -256
+}
+
+Test-Case: output
+Description: Test unary complement expressions with Unsigned16 operand.
+Labels: int:Unsigned16, int:UNSIGNED16_MAX_VALUE, unary-complement
+
+public function main() {
+    int:Unsigned16 unsignedInt = 40123;
+    int res = ~unsignedInt;
+    io:println(res); // @output -40124
+
+    io:println(~int:UNSIGNED16_MAX_VALUE);  // @output -65536
+}
+
+Test-Case: output
+Description: Test unary complement expressions with Signed16 operand.
+Labels: int:Signed16, int:SIGNED16_MAX_VALUE, unary-complement
+
+public function main() {
+    int:Signed16 signedInt = 12345;
+    int res = ~signedInt;
+    io:println(res); // @output -12346
+
+    io:println(~int:SIGNED16_MAX_VALUE); // @output -32768
+}
+
+Test-Case: output
+Description: Test unary complement expressions with Unsigned32 operand.
+Labels: int:Unsigned32, int:UNSIGNED32_MAX_VALUE, unary-complement
+
+public function main() {
+    int:Unsigned32 unsignedInt = 12034;
+    int res = ~unsignedInt;
+    io:println(res); // @output -12035
+
+    io:println(~int:UNSIGNED32_MAX_VALUE);  // @output -4294967296
+}
+
+Test-Case: output
+Description: Test unary complement expressions with Signed8 operand.
+Labels: int:Signed8, int:SIGNED8_MIN_VALUE, int:SIGNED8_MAX_VALUE, unary-complement
+
+public function main() {
+    int:Signed8 signedInt = 100;
+    int res = ~signedInt;
+    io:println(res); // @output -101
+
+    io:println(~int:SIGNED8_MIN_VALUE);  // @output 127
+    io:println(~int:SIGNED8_MAX_VALUE);  // @output -128
+}
+
+Test-Case: output
+Description: Test unary complement expressions with Signed16 operand.
+Labels: int:Signed16, int:SIGNED16_MIN_VALUE, int:SIGNED16_MAX_VALUE, unary-complement
+
+public function main() {
+    int:Signed16 a3 = 12345;
+    int res = ~a3;
+    io:println(res); // @output -12346
+
+    io:println(~int:SIGNED16_MIN_VALUE);  // @output 32767
+    io:println(~int:SIGNED16_MAX_VALUE);  // @output -32768
+}
+
+Test-Case: output
+Description: Test unary complement expressions with Signed32 operand.
+Labels: int:Signed32, int:SIGNED32_MIN_VALUE, int:SIGNED32_MAX_VALUE, unary-complement
+
+public function main() {
+    int:Signed32 a3 = 1000;
+    int res = ~a3;
+    io:println(res); // @output -1001
+
+    io:println(~int:SIGNED32_MIN_VALUE);  // @output 2147483647
+    io:println(~int:SIGNED32_MAX_VALUE);  // @output -2147483648
+}
+
+Test-Case: error
+Description: Test unary complement expression assignments resulting in an error.
+Labels: byte, int:Unsigned8, int:Unsigned16, int:Unsigned32, int:Signed8, int:Signed16,langlib, int:Signed32,
+module-type-defn, unary-complement
+
+type Ints -1|-2;
+
+public function main(byte x1, int:Unsigned8 x2, int:Unsigned16 x3, int:Unsigned32 x4, int:Signed8 x5,
```

Comment:
Shall we duplicate this test case for optional types also, to check the static type of result is optional int

Reply:
this is added